UserDataDownloadSubOrchestrator/handler.ts

```diff
@@ -0,0 +1,161 @@
+import { ActivityResultSuccess as SendUserDataDownloadMessageActivityResultSuccess } from "../SendUserDataDownloadMessageActivity/handler";
+import { ActivityResultSuccess as SetUserDataProcessingStatusActivityResultSuccess } from "../SetUserDataProcessingStatusActivity/handler";
+import { ProcessableUserDataProcessing } from "../UserDataDownloadOrchestrator/handler";
+
+const logPrefix = "UserDataDownloadSubOrchestrator";
+
+export type InvalidInputFailure = t.TypeOf<typeof InvalidInputFailure>;
+export const InvalidInputFailure = t.interface({
+  kind: t.literal("INVALID_INPUT"),
+  reason: t.string
+});
+
+export type UnhanldedFailure = t.TypeOf<typeof UnhanldedFailure>;
+export const UnhanldedFailure = t.interface({
+  kind: t.literal("UNHANDLED"),
+  reason: t.string
+});
+
+export type ActivityFailure = t.TypeOf<typeof ActivityFailure>;
+export const ActivityFailure = t.intersection([
+  t.interface({
+    activityName: t.string,
+    kind: t.literal("ACTIVITY"),
+    reason: t.string
+  }),
+  t.partial({ extra: t.object })
+]);
+
+export type OrchestratorFailure = t.TypeOf<typeof OrchestratorFailure>;
+export const OrchestratorFailure = t.taggedUnion("kind", [
+  InvalidInputFailure,
+  UnhanldedFailure,
+  ActivityFailure
+]);
+
+export type OrchestratorSuccess = t.TypeOf<typeof OrchestratorSuccess>;
+export const OrchestratorSuccess = t.interface({
+  kind: t.literal("SUCCESS")
+});
+
+export type SkippedDocument = t.TypeOf<typeof SkippedDocument>;
+export const SkippedDocument = t.interface({
+  kind: t.literal("SKIPPED")
+});
+
+export type OrchestratorResult = t.TypeOf<typeof OrchestratorResult>;
+export const OrchestratorResult = t.union([
+  OrchestratorFailure,
+  SkippedDocument,
+  OrchestratorSuccess
+]);
+
+const toActivityFailure = (
+  err: t.Errors,
+  activityName: string,
+  extra?: object
+) =>
+  ActivityFailure.encode({
+    activityName,
+    extra,
+    kind: "ACTIVITY",
+    reason: readableReport(err)
+  });
+
+export const handler = function*(
+  context: IFunctionContext,
+  document: unknown
+): IterableIterator<unknown> {
+  // This check has been done on the parent orchestrator, so it should never fail.
+  // However, it's maybe woth the effort of check it twice
+  const invalidInputOrCurrentUserDataProcessing = ProcessableUserDataProcessing.decode(
+    document
+  ).mapLeft<InvalidInputFailure>(err => {
+    context.log.error(
+      `${logPrefix}|WARN|Cannot decode ProcessableUserDataProcessing document: ${readableReport(
+        err
+      )}`
+    );
+    return InvalidInputFailure.encode({
+      kind: "INVALID_INPUT",
+      reason: readableReport(err)
+    });
+  });
+
+  if (isLeft(invalidInputOrCurrentUserDataProcessing)) {
+    return invalidInputOrCurrentUserDataProcessing.value;
+  }
+
+  const currentUserDataProcessing =
+    invalidInputOrCurrentUserDataProcessing.value;
+
+  // start this operation tomorrow
+  yield context.df.createTimer(new Date(Date.now() + 24 * 60 * 60 * 100));
```

Comment:
you cannot use Date.now() inside an orchestrator, you should use context.df.currentUtcDateTime instead, see

https://github.com/pagopa/io-functions-bonus/blob/master/EligibilityCheckOrchestrator/index.ts#L229

and https://docs.microsoft.com/it-it/azure/azure-functions/durable/durable-functions-code-constraints